htdocs/emailcollector/class/emailcollector.class.php

```diff
@@ -1904,6 +1904,7 @@ public function doCollectOneCollector()
 						}
 					} elseif ($operation['type'] == 'recordjoinpiece') {
 						$pj = getAttachments($imapemail, $connection);
+						$data = array();
```

Comment:
In method  executeHooks, the third param where $data is used is declared as
@param		Object	$object			Object to use hooks on
What happen if you set here
$data = new stdClass();
instead of 
$data = array();
Does is solve your need ? if yes it would be better...

Reply:
Ok no problem, $data could store a list of attachment as a array or as an object with list of entries